### PR TITLE
build: add skipRange and replaces to the rook csv

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
Add skipRange annotation and replaces field to spec in rook csv if the respective env variables are set.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by subhamkrai — APPROVED** [review]

**Review by subhamkrai — CHANGES_REQUESTED** [review]
@Nikhil-Ladha do you think we should wait for the downstream build to test on the branch so that if we need any extra changes we can do in this pr only?

File: build/csv/csv-gen.sh [hunk 1]
```diff
@@ -53,6 +53,14 @@ function generate_csv() {
     "${YQ_CMD_WRITE[@]}" "$CSV_FILE_NAME" metadata.annotations.externalClusterScript "$(base64 <$CEPH_EXTERNAL_SCRIPT_FILE)"
     "${YQ_CMD_WRITE[@]}" "$CSV_FILE_NAME" metadata.name "rook-ceph-operator.v${CSV_VERSION}"
 
+    if [[ -n $SKIP_RANGE ]]; then
```
Comment (travisn) [inline build/csv/csv-gen.sh:56]: Will SKIP_RANGE and REPLACES_CSV_VERSION have different settings in different downstream builds?

Comment (Nikhil-Ladha) [reply build/csv/csv-gen.sh:56]: Yes, the values will get updated for each release and are managed by the downstream build.

**Review by b-ranto — APPROVED** [review]
lgtm

**Review by subhamkrai — APPROVED** [review]

**Comment by subhamkrai** [comment]
> @Nikhil-Ladha do you think we should wait for the downstream build to test on the branch so that if we need any extra changes we can do in this pr only?

Based on the discuss we are good going ahead with the PR. Thanks

**Pull request merged**